include/revng/PipelineC/PipelineC.h

```diff
@@ -0,0 +1,378 @@
+#pragma once
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#ifdef __cplusplus
+#include <cstdint>
+
+#include "revng/Pipeline/Container.h"
+#include "revng/Pipeline/Kind.h"
+#include "revng/Pipeline/Runner.h"
+#include "revng/Pipeline/Step.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Pipes/PipelineManager.h"
+#else
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+#ifdef __cplusplus
+typedef revng::pipes::PipelineManager rp_runner;
```

Comment:
This is confusing. Shall we have `rp_pipeline_manager` or simply `rp_pipeline`?

Can you review the name of the C stuff to be as coherent as possible with C++ stuff? Otherwise we're making the life of newcomers approaching the C API harder for no reason.

In particular, this is not great:
```
 typedef const pipeline::TargetsList rp_container_status;
```